### PR TITLE
feat(*) add protobuf deps

### PR DESCRIPTION
**Pull request by gszr**
* Add protobuf libaries as package dependencies
* Add smoke test to ensure Kong's .proto file is included 
* Fix issue in tests where only API tests were being executed

### CODE REVIEW

File: test/tests/01-package/run.sh [hunk 1]
```diff
@@ -101,6 +97,11 @@ docker run ${USE_TTY} --user=root --rm ${KONG_TEST_IMAGE_NAME} /bin/sh -c "luaro
 docker run ${USE_TTY} --user=root --rm ${KONG_TEST_IMAGE_NAME} /bin/sh -c "luarocks config | grep -q /usr/local/openresty/luajit/lib"
 docker run ${USE_TTY} --user=root --rm ${KONG_TEST_IMAGE_NAME} /bin/sh -c "luarocks install version"
 
+# kong shipped files
+docker run ${USE_TTY} --user=root --rm ${KONG_TEST_IMAGE_NAME} /bin/sh -c "ls -l /etc/kong/kong.conf.default"
+docker run ${USE_TTY} --user=root --rm ${KONG_TEST_IMAGE_NAME} /bin/sh -c "ls -l /etc/kong/kong.logrotate"
+docker run ${USE_TTY} --user=root --rm ${KONG_TEST_IMAGE_NAME} /bin/sh -c "ls -l /usr/local/kong/lib/pluginsocket.proto"
+
```
Comment (hutchic) [inline test/tests/01-package/run.sh:104]: the changes to this file don't look wrong but seem somewhat unrelated?

File: build-kong.sh [hunk 2]
```diff
@@ -71,6 +71,10 @@ pushd /kong
   cp kong.conf.default /tmp/build/usr/local/lib/luarocks/rock*/kong/$ROCKSPEC_VERSION/
   cp kong.conf.default /tmp/build/etc/kong/kong.conf.default
   cp kong/pluginsocket.proto /tmp/build/usr/local/kong/lib
+
+  curl -fsSLo /tmp/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v3.19.0/protoc-3.19.0-linux-x86_64.zip
```
Comment (gszr) [inline build-kong.sh:75]: @javierguerragiraldez unsure about the version here. An idea is to pick it from the respective go-pdk version's `go.mod` file; what do you think?

**Comment by gAmUssA** [comment]
@gszr In which image tag this will be included?
